assets/js/base/context/cart-checkout/validation/index.js

```diff
@@ -132,9 +142,14 @@ export const ValidationContextProvider = ( { children } ) => {
 	 *                           value to false.
 	 */
 	const showValidationError = ( property ) => {
-		updateValidationError( property, {
-			hidden: false,
-		} );
+		if (
+			validationErrors[ property ] &&
+			validationErrors[ property ].hidden !== false
+		) {
+			updateValidationError( property, {
+				hidden: false,
+			} );
+		}
 	};
```

Comment:
`updateValidationError` already does a check for `property` on `validationErrors`. I'm guessing you added this logic to handle only updating if hidden is something different. I think we should move this logic check into `updateValidationError` (do a `shallowEqual` on the values to determine whether to update the state?) so that we can trust state is only going to be updated if necessary (avoiding this kind of gotcha anywhere else `updateValidationError` might be used).  Note, I do realize this is partial updates that are merged with the existing state, however, I still think we should do shallowEqual comparison on the object partials.

Reply:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2374/commits/e227e5e22c5f2d1db13b37ace1c0b1f35e6693ad